### PR TITLE
chore(litmus-portal): Projects integration and other fixes.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
Signed-off-by: ishangupta-ds <ishan.gupta@mayadata.io>

## Proposed changes

Integration and bug fixes for Projects, workflow manifests and optimization of state-management.

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Dependency
- None.

## Special notes for your reviewer:
High Priority PR.

### CODE REVIEW

**Comment by ishangupta-ds** [comment]
@rajdas98 @gdsoumya @arkajyotiMukherjee @Jonsy13 @Sanjay1611 PTAL.

File: litmus-portal/frontend/package.json [hunk 1]
```diff
@@ -21,9 +21,6 @@
     "brace": "^0.11.1",
     "date-fns": "^2.14.0",
     "graphql": "^15.3.0",
-    "i18next": "^19.7.0",
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/package.json:24]: why was i18 removed @ishangupta-ds 
@chandankumar4 was this something expected?

File: litmus-portal/frontend/src/components/Sections/ChaosWorkflows/BrowseWorkflow/index.tsx [hunk 2]
```diff
@@ -59,20 +62,24 @@ interface SortData {
 const BrowseWorkflow = () => {
   const classes = useStyles();
 
+  const userData: UserData = useSelector((state: RootState) => state.userData);
+
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/ChaosWorkflows/BrowseWorkflow/index.tsx:66]: remove additional line after each variable declaration

File: litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx [hunk 3]
```diff
@@ -19,7 +21,21 @@ const PersonalDetails: React.FC = () => {
 
   const userData: UserData = useSelector((state: RootState) => state.userData);
 
-  const { name, email, username } = userData;
+  const { username } = userData;
+
+  let email = '';
+
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/Settings/AccountsTab/PersonalDetails/index.tsx:27]: here too

File: litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/UserManagement/index.tsx [hunk 4]
```diff
@@ -237,7 +236,7 @@ const UserManagement: React.FC = () => {
                       </TableRow>
                     </TableHead>
                     <TableBody>
-                      {rows ? (
+                      {rows.length > 0 ? (
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Sections/Settings/UserManagementTab/UserManagement/index.tsx:239]: just use rows.length
0 is falsey value already

File: litmus-portal/frontend/src/components/WelcomeModal/Modalpage.tsx [hunk 5]
```diff
@@ -21,13 +20,12 @@ const ModalPage: React.FC<ModalDataProps> = ({
   setText,
   setName,
 }) => {
-  const { t } = useTranslation();
   const classes = useStyles();
   return (
     <div className={classes.insideModal}>
       <Icon />
       <div className={classes.heading}>
-        {t('welcomeModel.title')}
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/WelcomeModal/Modalpage.tsx:30]: why as all translation changes are overridden?

File: litmus-portal/frontend/src/containers/layouts/Header/index.tsx [hunk 6]
```diff
@@ -4,20 +4,36 @@ import Toolbar from '@material-ui/core/Toolbar';
...
   const userData: UserData = useSelector((state: RootState) => state.userData);
 
-  const { name, email, username } = userData;
+  const { username } = userData;
+
+  let email = '';
+
+  let name = '';
+
+  useQuery(GET_USER, {
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/containers/layouts/Header/index.tsx:28]: query and query variables need to be typed in models folder

File: litmus-portal/frontend/src/pages/HomePage/index.tsx [hunk 7]
```diff
@@ -46,10 +45,21 @@ const CreateWorkflowCard = () => {
 
 const HomePage = () => {
   const [isOpen, setIsOpen] = useState<boolean>(true);
+
   const { userData } = useSelector((state: RootState) => state);
-  const { name } = userData;
+
+  let name = '';
+
+  useQuery(GET_USER, {
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/pages/HomePage/index.tsx:53]: same here

File: litmus-portal/frontend/src/pages/WorkflowUnderground/index.tsx [hunk 8]
```diff
@@ -21,12 +23,16 @@ interface WorkflowUndergroundProps {
 const WorkflowUnderground: React.FC<WorkflowUndergroundProps> = ({
   location,
 }) => {
-  const { t } = useTranslation();
   const classes = useStyles();
+
+  const userData: UserData = useSelector((state: RootState) => state.userData);
+
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/pages/WorkflowUnderground/index.tsx:29]: remove empty space

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]
@chandankumar4 I want you to review this too

File: litmus-portal/frontend/package.json (hunk 1 above)
Comment (chandankumar4) [reply litmus-portal/frontend/package.json:24]: We need this!

File: litmus-portal/frontend/src/containers/layouts/Header/index.tsx (hunk 6 above)
Comment (arkajyotiMukherjee) [reply litmus-portal/frontend/src/containers/layouts/Header/index.tsx:28]: take reference from workflowData.ts in models folder @ishangupta-ds

File: litmus-portal/frontend/src/graphql/index.ts [hunk 9]
```diff
@@ -57,16 +57,23 @@ export const GET_USER = gql`
...
-    }
-  }
-`;
-
-export const GET_PROJECT = gql`
-  query getProject($projectID: String!) {
-    getProject(projectID: $projectID) {
-      name
-      id
+      projects {
```
Comment (oumkale) [inline litmus-portal/frontend/src/graphql/index.ts:60]: @ishangupta-ds You took projects, please make it project.

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts [hunk 10]
```diff
@@ -5,10 +5,10 @@ export default [
     urlToIcon:
       'https://hub.litmuschaos.io/api/icon/1.7.0/generic/node-cpu-hog.png',
     chaosWkfCRDLink:
-      'https://raw.githubusercontent.com/litmuschaos/chaos-charts/master/workflows/node-cpu-hog/workflow.yaml',
+      'https://raw.githubusercontent.com/ishangupta-ds/chaos-workflows/master/Argo/argowf-native-node-cpu-hog.yaml',
 
     gitLink:
-      'https://github.com/litmuschaos/chaos-charts/blob/master/workflows/node-cpu-hog/workflow.yaml',
+      'https://github.com/ishangupta-ds/chaos-workflows/blob/master/Argo/argowf-native-node-cpu-hog.yaml',
```
Comment (gdsoumya) [inline litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:11]: Change to litmus repo

File: litmus-portal/backend/auth/go.sum [hunk 11]
```diff
@@ -201,6 +213,7 @@ gopkg.in/go-playground/assert.v1 v1.2.1/go.mod h1:9RXL0bg/zibRAgZUYszZSwO/z8Y/a8
 gopkg.in/go-playground/validator.v9 v9.29.1/go.mod h1:+c9/zcJMFNgbLvly1L1V+PpxWdVbfP1avr/N00E2vyQ=
 gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.4/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gopkg.in/yaml.v2 v2.2.8 h1:obN1ZagJSUGI0Ek/LBmuj4SNLPfIny3KsKFopxRdj10=
 gopkg.in/yaml.v2 v2.2.8/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 sourcegraph.com/sourcegraph/appdash v0.0.0-20180110180208-2cc67fd64755/go.mod h1:hI742Nqp5OhwiqlzhgfbWU4mW4yO10fP+LoT9WOswdU=
 sourcegraph.com/sourcegraph/appdash-data v0.0.0-20151005221446-73f23eafcf67/go.mod h1:L5q+DGLGOQFpo1snNEkLOJT2d1YTW66rWNzatr3He1k=
```
Comment (gdsoumya) [inline litmus-portal/backend/auth/go.sum:219]: Remove these changes

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts [hunk 12]
```diff
@@ -20,13 +20,58 @@ export default [
...
+    totalRuns: 4300,
+    isCustom: false,
+  },
+  {
+    workflowID: '3',
+    title: 'pod-cpu-hog',
+    urlToIcon:
+      'https://hub.litmuschaos.io/api/icon/1.7.0/generic/pod-cpu-hog.png',
+    chaosWkfCRDLink:
+      'https://raw.githubusercontent.com/ishangupta-ds/chaos-workflows/master/Argo/argowf-native-pod-cpu-hog.yaml',
```
Comment (gdsoumya) [inline litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts:38]: Change all links to litmus repo

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: litmus-portal/frontend/package.json (hunk 1 above)
Comment (gdsoumya) [reply litmus-portal/frontend/package.json:24]: I think @ishangupta-ds did not pull the new changes or rejected the incoming changes.

File: litmus-portal/backend/auth/go.sum [hunk 13]
```diff
@@ -13,19 +13,25 @@ github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSs
 github.com/dgrijalva/jwt-go v3.2.0+incompatible h1:7qlOGliEKZXTDg6OTjfoBKDXWrumCAMpl/TFQ4/5kLM=
```
Comment (oumkale) [inline litmus-portal/backend/auth/go.sum:13]: Why those changes are here. Will you remove changes from go.sum

File: litmus-portal/frontend/src/graphql/index.ts (hunk 9 above)
Comment (gdsoumya) [reply litmus-portal/frontend/src/graphql/index.ts:60]: It is `projects`, it has been updated in the server

File: litmus-portal/frontend/package.json (hunk 1 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/package.json]: No i did pull the changes, an merged my code, as all translations were just delete and there were no conflicts so this happened, had to reset to @chandankumar4  's commit. Will raise a new PR shortly with the same changes and translation files included. Incorporating the suggested changes to this PR @chandankumar4 @arkajyotiMukherjee @gdsoumya @oumkale cc: @rajdas98

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 10 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts]: Cant change this @gdsoumya till i raise a PR from my repo to litmuschaos org cc: @ksatchit @rajdas98

File: litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts (hunk 12 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/components/PredifinedWorkflows/data.ts]: It is not possible to do that as we don't have these in litmuschaos org. @ksatchit will be changing 2 of the templates then we will raise a PR for the same. We can change it after that. @gdsoumya

File: litmus-portal/backend/auth/go.sum (hunk 13 above)
Comment (ishangupta-ds) [reply litmus-portal/backend/auth/go.sum]: These changes came in on starting the server @oumkale

File: litmus-portal/backend/auth/go.sum (hunk 11 above)
Comment (ishangupta-ds) [reply litmus-portal/backend/auth/go.sum]: These changes came in on starting the server @gdsoumya

**Pull request closed without merging**